packages/next/client/script.tsx

```diff
@@ -82,19 +85,26 @@ const loadScript = (props: ScriptProps): void => {
   if (src) {
     ScriptCache.set(src, loadPromise)
   }
```

Comment:
Can we move this to the `else if (src) {` below so it's cleaner?

Reply:
@shuding Sure!

BTW, do you have any suggestions about how to refactor the `beforeInteractive` inline script?

Currently, `loadScript` is called directly in the render phase for `worker` and `beforeInteractive`, making it react strict mode in-compatible. When I enable `reactStrictMode` to validate my fix, other test cases begin to fail.

Reply:
Updated in [`f7669de` (#40002)](https://github.com/vercel/next.js/pull/40002/commits/f7669de5231b39ca8e4aec57a7075f6388b0507a).